server/BraveAlerterConfigurator.js

```diff
@@ -103,6 +104,10 @@ class BraveAlerterConfigurator {
           session.fallBackAlertTwilioStatus = alertSession.fallbackReturnMessage
         }
 
+        if (!session.respondedAt && session.state !== ALERT_STATE.STARTED && session.state !== ALERT_STATE.WAITING_FOR_REPLY) {
+          session.respondedAt = await db.getCurrentTime(client)
+        }
+
```

Comment:
I'm not super happy with this `if`-statement. It looks like the other ones above it, but because it depends on `session.state` which could have changed in an earlier `if`-statement, it's sort of different. The value of `session.state` at this point should be the new State, not the old one, whereas up until this point, I'd been using old values from `session` and new values from `alertSession`. However, I can't use `alertSession` here because sometimes it's not sent because it's not being updated (for example, when we update the Twilio Fallback Status, we don't change the Session State). Although I think we've agreed that we can remove the Twilio Fallback Status column and no longer update it, but this doesn't seem to be the time to do that.

Reply:
I was trying to cover too many edge cases, let's change this to:
```
if (alertSession.alertState !== ALERT_STATE.WAITING_FOR_CATEGORY) {          
   session.respondedAt = await db.getCurrentTime(client)        
}
```

Reply:
that looks great, it's much easier to follow now.

Reply:
Fixed. Thanks for working through this one with me @schwarrrtz :smile:

Reply:
no worries! :)